### ISSUE TITLE
Headings disappear if using ng-if on ancestor element

### ISSUE THREAD

**Issue by johnrix** [daniel-nagy/fixed-table-header#31]
The following causes the headings to become invisible:

    <div ng-if="1">
	    <div style="overflow: auto; max-height: 300px;">
		<table>
			<thead fix-head>
				<tr><th>Name</th></tr>
			</thead>
			<tbody>
				<tr><td>A</td></tr>
				<tr><td>A</td></tr>
			</tbody>
	  	</table>
	    </div>
    </div>

Removing the ng-if restores them.

**Comment by skortchmark9** [comment]
As a short term workaround you can use ng-show instead.

**Comment by johnrix** [comment]
Yep, I did confirm that works at least. For some reason though, the column headings were not aligning for me, so I since gave up on using the directive. I suspect it may have been to do with me using ng-repeat on the `<th>` elements (I have a variable set of columns, according to a selected report).

**Comment by skortchmark9** [comment]
I don't think so, it's because the elements are all hidden when the math is done to determine their widths. As a second workaround you can do
```javascript
$timeout(function() {
  // force table header to recalculate its width.
  window.dispatchEvent(new Event('resize'));
});
```

**Comment by johnrix** [comment]
Ah good point, you're more than likely right there!

**Comment by Sayaren** [comment]
use ng-show instead

**Comment by ceiger89** [comment]
in the ng-repeat directive, the same.
